modules/rgbd/src/tsdf.cpp

```diff
@@ -11,9 +11,53 @@
 namespace cv {
 
 namespace kinfu {
+/*
+static inline TsdfType floatToTsdf_INTR(float num)
+{
+    Cv32suf u;
+    u.f = num;
+    u.u += 7 << 23; // multiply by 128 == add 7 to exponent
+    u.u ^= 1u << 31; // negate
+    int tsdf = (int)u.f;
+    tsdf = tsdf ? tsdf : (num < 0 ? 1 : -1);
+    return tsdf >= 128 ? 127 : tsdf;
+}
+
+static inline float tsdfToFloat(TsdfType num)
+{
+    if(!num) return 0.f;
+    Cv32suf u;
+    u.f = float(num);
+    u.u -= 7 << 23; // divide by 128 == sub 7 from exponent
+    u.u ^= 1u << 31; // negate
+    return u.f;
+}
+*/
+
+static inline v_float32x4 tsdfToFloat_INTR(v_int32x4 num)
+{
+    //v_float32x4 num128 = v_setall_f32(-128);
+    //return v_cvt_f32(num) / num128;
+    v_float32x4 num128 = v_setall_f32(-0.0078125f); // = (-1/128)
+    return v_cvt_f32(num) * num128;
+}
+
+static inline TsdfType floatToTsdf(float num)
+{
+    //CV_Assert(-1 < num <= 1);
+    int8_t res = int8_t(num * (-128));
+    res = res ? res : (num < 0 ? 1 : -1);
+    return res;
+}
+
+static inline float tsdfToFloat(TsdfType num)
+{
+    //return float(num) / (-128);
+    return float(num) * (-0.0078125f); // * (-1/128)
```

Comment:
The same notion about constants as in `hash_tsdf.cpp` file

Reply:
done